climakitae/explore/amy.py

```diff
@@ -203,6 +291,609 @@ def _return_diurnal(y: xr.DataArray) -> xr.DataArray:
+                            future_profile[col]
+                            - historic_profile.iloc[:, historic_col_idx]
+                        )
+                pbar.update(1)
+
+    else:
+        # Both have single-level columns
+        # Check if columns match
+        if list(future_profile.columns) == list(historic_profile.columns):
+            print("   ✓ Columns match - computing element-wise difference")
+            difference_profile = future_profile - historic_profile
+        else:
+            print("   ⚠️  Warning: Column mismatch between future and historic profiles")
+            print(f"      Future columns: {list(future_profile.columns)[:5]}...")
+            print(f"      Historic columns: {list(historic_profile.columns)[:5]}...")
+            # Try to align by position
+            min_cols = min(len(future_profile.columns), len(historic_profile.columns))
+            difference_profile = (
+                future_profile.iloc[:, :min_cols] - historic_profile.iloc[:, :min_cols]
+            )
+
+    print(
+        f"✅ Climate profile computation complete! Final shape: {difference_profile.shape}"
+    )
+    print(
+        f"   (Days: {difference_profile.shape[0]}, Hours/Columns: {difference_profile.shape[1]})"
+    )
+    return difference_profile
+
+
+def compute_profile(data: xr.DataArray, days_in_year: int = 365, q=0.5) -> pd.DataFrame:
+    """
+    Calculates the average meteorological year profile for warming level data using 8760
+    analysis.
+
+    This function handles global warming levels approach using time_delta coordinate.
+    Takes the first 8760 hours (one year) from the time_delta dimension and processes
+    each warming level and simulation separately, preserving both dimensions.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+        Hourly base-line subtracted data for one variable with warming_level,
+        time_delta, and simulation dimensions
+
+    days_in_year : int, optional
+        Either 366 or 365, depending on whether or not the year is a leap year.
+        Default to 365 days
+
+    q : float, optional
+        Quantile value for selecting representative values. Currently unused
+        in the optimized implementation.
+
+    Returns
+    -------
+    pd.DataFrame
+        Average meteorological year table for each warming level and simulation,
+        with days of year as the index and hour of day as the columns.
+        If multiple warming levels and/or simulations exist, they will be
+        included as additional column levels.
+
+    """
+    # Check for simulation dimension
+    has_simulation = "simulation" in data.dims
+    if has_simulation:
+        n_simulations = len(data.simulation)
+        simulations = data.simulation.values
+    else:
+        n_simulations = 1
+        simulations = [None]
+
+    # Slice to first 8760 hours (one year) from time_delta
+    data_8760 = data.isel(time_delta=slice(0, 8760))
```

Comment:
@vicford We are just taking the _**first**_ 8760 of historical and future WL data (i.e. for WL 1.5 with centered_year of 2030, this will just be data from the year 2015), and subtracting them to make the climate profile